_posts/2019-03-18-openfaas-on-nomad.md

```diff
@@ -0,0 +1,153 @@
+
+![openfaas-fargate](/images/openfaas-on-nomad/openfaas_nomad_vault.jpg)
+
+Let's look at the Vault integration more closely:
+
+### Vault faas-nomad startup arguments
+
+| arg  |  description |
+|---|---|
+| vault_addr | default from Nomad agent. Overrides the URL of the Vault service. |
+| vault_default_policy | default: openfaas. The name of the Vault policy to limit OpenFaaS' access to Vault. |
+| vault_secret_path_prefix | default: secret/openfaas. The preconfigured kv v1 secrets backend path. |
+| vault_app_role_id | default: empty. The preconfigured app roll ID used with the Vault approle backend. |
+| vault_app_secret_id | default: blank. The preconfigured app roll secret ID used with the approle backend. |
+| vault_tls_skip_verify | default: false. Skips TLS verification for calls to Vault. Not recommend in production. |
+
+Let's provision a Vault instance to handle the secrets API of OpenFaaS:
+
+Reference example:
+[https://raw.githubusercontent.com/hashicorp/faas-nomad/master/provisioning/scripts/vault_populate.sh](https://raw.githubusercontent.com/hashicorp/faas-nomad/master/provisioning/scripts/vault_populate.sh)
+
+1) Enable the approle auth backend in Vault:
+
+```shell
+vault auth enable approle
+```
+
+2) Create a policy for faas-nomad and OpenFaaS functions:
+   
+```shell
+vault policy write openfaas policy.hcl
+```
+   
+   Policy file example: [https://raw.githubusercontent.com/hashicorp/faas-nomad/master/provisioning/scripts/policy.hcl](https://raw.githubusercontent.com/hashicorp/faas-nomad/master/provisioning/scripts/policy.hcl)
+
+   It is important that the policy contain: create, update, delete and list capabilities that match your secret backend prefix. In this case, path "secret/openfaas/*" will work with the default configuration.
+
+   Also, faas-nomad takes care of renewing it's own auth token, so we need to make sure the policy uses path "auth/token/renew-self" and has the "update" capability.
+
+3) Setup the approle itself:  
+```shell
+curl -i \
+  --header "X-Vault-Token: ${VAULT_TOKEN}" \
+  --request POST \
+  --data '{"policies": ["openfaas"], "period": "24h"}' \
+  https://${VAULT_HOST}/v1/auth/approle/role/openfaas
+```
+   This creates the role attached to the policy we just created. The "period" property and duration is important for renewing long-running service Vault tokens.
+```shell
+curl -i \
+  --header "X-Vault-Token: ${VAULT_TOKEN}" \
+  https://${VAULT_HOST}/v1/auth/approle/role/openfaas/role-id
+```
+Produces the role_id needed for -vault_app_role_id cli argument.
+```shell
+curl -i \
+  --header "X-Vault-Token: ${VAULT_TOKEN}" \
+  --request POST \
+  https://${VAULT_HOST}/v1/auth/approle/role/openfaas/secret-id
+```
+   Produces the secret_id needed for -vault_app_secret_id cli argument.
+
+Let's assume the Vault parameters have been populated in the run args, and you're now running faas-nomad along with the other OpenFaaS components. Now, try out the new faas-cli secret commands:
```

Comment:
Maybe list how you would execute faas-nomad? Is this using a `job specification` in Nomad